src/components/paginator.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+
+import { Dispatch } from '../lib/redirects-manager-state';
+
+type getPageNumbersType = (viewLength: number, perPage: number) => number[];
+
+type validateSelectedPageType = (input: string, max: number) => number;
+
+interface PaginatorComponentProps {
+    viewLength: number;
+    filterBy: string;
+    storeLength: number;
+    perPage: number;
+    currentPage: number;
+    dispatch: Dispatch;
+}
+
+interface PageSelectorProps { pageNumbers: number[]; current: number; dispatch: Dispatch; }
+
+const getPageNumbers: getPageNumbersType = (viewLength, perPage) => {
+    const pageNumbers = [];
+    const calculatedPerPage = perPage ? perPage : viewLength;
+    for (let i = 1; i <= Math.ceil(viewLength / calculatedPerPage); i++) {
+        pageNumbers.push(i);
+    }
+    return pageNumbers;
+};
+
+const validateSelectedPage: validateSelectedPageType = (input, max) => {
+    const page = Number(input);
+    if (page < 1) { return 1; }
+    if (page > max) { return max; }
+    return page;
+};
+
+const PageSelector = ( { pageNumbers, current, dispatch }: PageSelectorProps) => {
+    const currentPage = current ? current : 1;
+    const lastPage = pageNumbers[pageNumbers.length - 1];
+    return (
+        <div className="alignright actions" style={{display: 'flex'}}>
+            <span  style={{ marginTop: '6px'}}>Page </span>
+            <input type="number"
```

Comment:
looks like you're only offering a simple text-input for pagination, which usually is very cumbersome to use. please offer a button-based pagination bar. something more like this: https://reactjsexample.com/a-reactjs-dumb-component-to-render-a-pagination/

Reply:
Lets discuss this on voice

Reply:
applied